src/test/java/uk/gov/pay/api/validation/PaymentOutcomeValidatorTest.java

```diff
@@ -0,0 +1,150 @@
+
+        assertThat(constraintViolations.size(), is(1));
+        assertThat(constraintViolations.iterator().next().getMessage(), is("Field [payment_outcome] must include a valid status and error code"));
+    }
+
+    @Test
+    public void failsValidationForInvalidErrorCode() {
+
+        PaymentOutcome paymentOutcome = new PaymentOutcome(
+                "failed",
+                "error",
+                new Supplemental(
+                        "ECKOH01234",
+                        "textual message describing error code"
+                )
+        );
+
+        CreateTelephonePaymentRequest telephonePaymentRequest = telephoneRequestBuilder
+                .paymentOutcome(paymentOutcome)
+                .build();
+
+        Set<ConstraintViolation<CreateTelephonePaymentRequest>> constraintViolations = validator.validate(telephonePaymentRequest);
+
+        assertThat(constraintViolations.size(), is(1));
+        assertThat(constraintViolations.iterator().next().getMessage(), is("Field [payment_outcome] must include a valid status and error code"));
+    }
+
+    @Test
+    public void passesValidationForCorrectErrorCode() {
+
+        PaymentOutcome paymentOutcome = new PaymentOutcome(
+                "failed",
+                "P0010",
+                new Supplemental(
+                        "ECKOH01234",
+                        "textual message describing error code"
+                )
+        );
+
+        CreateTelephonePaymentRequest telephonePaymentRequest = telephoneRequestBuilder
+                .paymentOutcome(paymentOutcome)
+                .build();
+
+        Set<ConstraintViolation<CreateTelephonePaymentRequest>> constraintViolations = validator.validate(telephonePaymentRequest);
+
+        assertThat(constraintViolations.isEmpty(), is(true));
+    }
+
+    @Test
+    public void passesValidationForPaymentOutcomeStatusOfSuccess() {
+
+        CreateTelephonePaymentRequest telephonePaymentRequest = telephoneRequestBuilder
+                .paymentOutcome(new PaymentOutcome("success"))
+                .build();
+
+        Set<ConstraintViolation<CreateTelephonePaymentRequest>> constraintViolations = validator.validate(telephonePaymentRequest);
+
+        assertThat(constraintViolations.isEmpty(), is(true));
+    }
+
+    @Test
+    public void passesValidationForNullPaymentOutcome() {
+
+        CreateTelephonePaymentRequest telephonePaymentRequest = telephoneRequestBuilder
+                .paymentOutcome(null)
+                .build();;
+
+        Set<ConstraintViolation<CreateTelephonePaymentRequest>> constraintViolations = validator.validate(telephonePaymentRequest);
+
+        assertThat(constraintViolations.size(), is(1));
+        assertThat(constraintViolations.iterator().next().getMessage().equals("Field [payment_outcome] must include a valid status and error code"), is(false));
+    }
```

Comment:
If this was at the `PaymentOutcome` level it would use
```
Set<ConstraintViolation<PaymentOutcome>> constraintViolations = validator.validate(telephonePaymentRequest);
```
and then it could be `passesValidationForNullPaymentOutcome` and the assertion would be
```
assertThat(constraintViolations.isEmpty(), is(true));
```
, but it's not so it should test what the actual constraint violation is (applies to all validation tests).
For other validator tests - they're not at the deepest validator level (which in some cases seem either impossible or difficult to do), and there is no general test for `CreateTelephonePaymentRequest`, so checking for the actual `not null` violation should be a part of them.

Reply:
Done - I've directly tested for not null message

Reply:
Not in `PaymentOutcomeValidatorTest`